development.md

```diff
@@ -0,0 +1,111 @@
+# The Compose Specification - Development support
+{:.no_toc}
+
+*Note:* Development support is an OPTIONAL part of the Compose Specification.
+
+* ToC
+{:toc}
+
+## Introduction
+
+Compose specification is a platform-neutral way to define multi-container applications. A Compose implementation
+focusing on development use-case to run application on local machine SHOULD also support some development hooks to improve developers velocity on their local workflow (also known as "inner loop"). The Compose Development specification allows to define how a Compose implementation SHOULD behave to efficiently assist the developer.
+
+## Definitions
+
+Compose Specification is extended to support an OPTIONAL `develop` subsection on services. This section
+defines the development constraints and workflows to be set by a Compose Implementation to assist the developer
+working on service codebase. Only a subset of Compose file services MAY define such a Development subsection.
+
+## Illustrative sample
+
+The following sample illustrates Compose specification concepts with a concrete sample application. The sample is non-normative.
+
+```yaml
+services:
+  frontend:
+    image: awesome/webapp
+    build: ./webapp
+    develop:
+      watch: 
+        ignores:
+            - node_modules/
+        quiet-period: 5s
+        trigger: 
+          # synchronize src with container's /app
+          strategy: sync
+          paths: 
+            - src:/app
+
+  backend:
+    image: awesome/backend
+    build: ./backend
+    develop:
+      watch: 
+        # rebuild image and recreate service
+        trigger: build
+
+  db:
+    image: awesome/database
+    develop:
+      watch: 
+        trigger:
+          strategy: sync
+          # get service to reload configuration
+          signal: SIGHUP
```

Comment:
Should this example include a path for what's being synced? According to the definition, build context would be used as a default but one is not defined here.